sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild

```diff
@@ -101,50 +97,42 @@ multilib_src_configure() {
 }
 
 multilib_src_compile() {
-	if ! multilib_is_native_abi || ! use tools ; then
+	if multilib_is_native_abi && use tools ; then
+		emake V=1
+	else
 		emake -C lib/et V=1
 		emake -C lib/ss V=1
-		if use tools ; then
-			emake -C lib/ext2fs V=1
-			emake -C lib/e2p V=1
-		fi
-		return 0
+		emake -C lib/ext2fs V=1
+		emake -C lib/e2p V=1
 	fi
-
-	emake V=1
 }
 
 multilib_src_test() {
-	if multilib_is_native_abi ; then
+	if multilib_is_native_abi && use tools ; then
 		emake V=1 check
 	else
 		# For non-native, there's no binaries to test. Just libraries.
+		emake -C lib/support V=1 # required by lib/ext2fs
 		emake -C lib/et V=1 check
 		emake -C lib/ss V=1 check
+		emake -C lib/ext2fs V=1 check
+		emake -C lib/e2p V=1 check
 	fi
 }
 
 multilib_src_install() {
-	if ! multilib_is_native_abi || ! use tools ; then
+	if multilib_is_native_abi && use tools ; then
+		emake STRIP=':' DESTDIR="${D}" install
+	else
 		emake -C lib/et V=1 DESTDIR="${D}" install
 		emake -C lib/ss V=1 DESTDIR="${D}" install
-
-		if use tools ; then
-			emake -C lib/ext2fs V=1 DESTDIR="${D}" install
-			emake -C lib/e2p V=1 DESTDIR="${D}" install
-		fi
-	else
-		emake \
-			STRIP=: \
-			DESTDIR="${D}" \
-			install
-
-		# Move shared libraries to /lib/, install static libraries to
-		# /usr/lib/, and install linker scripts to /usr/lib/.
-		gen_usr_ldscript -a e2p ext2fs
+		emake -C lib/ext2fs V=1 DESTDIR="${D}" install
+		emake -C lib/e2p V=1 DESTDIR="${D}" install
 	fi
 
-	gen_usr_ldscript -a com_err ss $(usex kernel_linux '' 'uuid blkid')
+	# Move shared libraries to /lib/, install static libraries to
+	# /usr/lib/, and install linker scripts to /usr/lib/.
+	gen_usr_ldscript -a com_err ss ext2fs e2p $(usex kernel_linux '' 'uuid blkid')
```

Comment:
usex can go actually, we don't support eg Gentoo/fbsd anymore, and prefix won't be concerned with split usr booting issues